src/renku_crc/server_options.py

```diff
@@ -0,0 +1,167 @@
+
+The purpose of this is to be able to create resource
+pools and classes based on the old server options until the admin UI interface
+is added.
+"""
+from typing import Any, List, Set
+
+from pydantic import BaseModel, ByteSize, Extra, Field, validator
+
+import models
+
+
+def _check_greater_than_zero(cls: Any, v: int | float) -> int | float:
+    if v <= 0:
+        raise ValueError(f"The provided value should be greater than zero, instead it was {v}.")
+    return v
+
+
+class ServerOptionsDefaults(BaseModel):
+    """Used to parse the server option defaults passed to the notebook service in the Helm values."""
+
+    cpu_request: float = Field(gt=0)
+    mem_request: ByteSize
+    disk_request: ByteSize
+    gpu_request: int = Field(ge=0, default=0)
+
+    class Config:
+        """Configuration."""
+
+        extra = Extra.ignore
+
+
+class _ServerOptionsCpu(BaseModel):
+    options: List[float] = Field(min_items=1)
+
+    class Config:
+        extra = Extra.ignore
+
+    @validator("options", pre=False, each_item=True)
+    def greater_than_zero(cls, val):
+        return _check_greater_than_zero(cls, val)
+
+
+class _ServerOptionsGpu(BaseModel):
+    options: List[int] = Field(min_items=1)
+
+    class Config:
+        extra = Extra.ignore
+
+    @validator("options", pre=False, each_item=True)
+    def greater_than_or_equal_to_zero(cls, v):
+        if v < 0:
+            raise ValueError(f"The provided value should be greater than or equal to zero, instead it was {v}.")
+        return v
+
+
+class _ServerOptionsBytes(BaseModel):
+    options: List[ByteSize] = Field(min_items=1)
+
+    class Config:
+        extra = Extra.ignore
+
+    @validator("options", pre=True)
+    def convert_units(cls, vals):
+        for ival, val in enumerate(vals):
+            if isinstance(val, str) and val.strip().endswith("i"):
+                vals[ival] = val.strip() + "b"
+        return vals
+
+    @validator("options", pre=False, each_item=True)
+    def greater_than_zero(cls, val):
+        return _check_greater_than_zero(cls, val)
+
+
+class ServerOptions(BaseModel):
+    """Used to parse the server options passed to the notebook service in the Helm values."""
+
+    cpu_request: _ServerOptionsCpu
+    mem_request: _ServerOptionsBytes
+    disk_request: _ServerOptionsBytes
+    gpu_request: _ServerOptionsGpu = Field(default_factory=lambda: _ServerOptionsGpu(options=[0]))
+
+    class Config:
+        """Configuration."""
+
+        extra = Extra.ignore
+
+    def find_largest_attribute(self) -> str:
+        """Find the attribute with the largest number of choices."""
+        max_elems = 0
+        largest_list = ""
+        for k, v in self.dict().items():
+            options = v.get("options", [])
+            if k == "disk_request":
+                continue
+            if len(options) > max_elems:
+                max_elems = len(options)
+                largest_list = k
+        return largest_list
```

Comment:
```suggestion
       options = ((k, v.get("options", [])) for k,v in self.dict().items() if k != "disk_request")
       largest_list = max(options, key=lambda t: len(t[1]))[0]
       return largest_list
```
a bit shorter, not that it really makes a difference though